src/nvim/spell.c

```diff
@@ -4507,40 +4497,32 @@ static void suggest_trie_walk(suginfo_T *su, langp_T *lp, char_u *fword, bool so
         break;
       }
 
-      if (has_mbyte) {
-        n = MB_CPTR2LEN(p);
-        c = mb_ptr2char(p);
-        if (p[n] == NUL)
-          c2 = NUL;
-        else if (!soundfold && !spell_iswordp(p + n, curwin))
-          c2 = c;           // don't swap non-word char
-        else
-          c2 = mb_ptr2char(p + n);
+      n = MB_CPTR2LEN(p);
+      c = utf_ptr2char(p);
+      if (p[n] == NUL) {
+        c2 = NUL;
+      } else if (!soundfold && !spell_iswordp(p + n, curwin)) {
+        c2 = c;  // don't swap non-word char
       } else {
-        if (p[1] == NUL)
-          c2 = NUL;
-        else if (!soundfold && !spell_iswordp(p + 1, curwin))
-          c2 = c;           // don't swap non-word char
-        else
-          c2 = p[1];
+        c2 = utf_ptr2char(p + n);
       }
 
       // When the second character is NUL we can't swap.
       if (c2 == NUL) {
         PROF_STORE(sp->ts_state)
         sp->ts_state = STATE_REP_INI;
         break;
-      }
+          }
```

Comment:
Indentation is wrong from here to the `#ifdef DEBUG_TRIEWALK`.

Reply:
Done.

Reply:
The new change shits the code to the right, the old indentation was okay.

Reply:
Done